Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/UpdateManager.java

```diff
@@ -227,25 +240,49 @@ private void runUpdateApi(final boolean manual) {
         }
     }
 
-    private boolean processUpdateApiResponse(UpdateApiResponse response) {
+    private boolean processUpdateApiResponse(UpdateApiResponse response, boolean manual) {
         if ((response.versionCode > BuildConfig.VERSION_CODE || BuildConfig.DEV_BUILD)
                 && BackgroundUtils.isInForeground()) {
-            boolean concat = !response.updateTitle.isEmpty();
-            CharSequence updateMessage =
-                    concat ? TextUtils.concat(response.updateTitle, "; ", response.body) : response.body;
-            AlertDialog dialog = new AlertDialog.Builder(context).setTitle(
-                    getApplicationLabel() + " " + response.versionCodeString + " available")
-                    .setMessage(updateMessage)
-                    .setNegativeButton(R.string.update_later, null)
-                    .setPositiveButton(R.string.update_install, (dialog1, which) -> updateInstallRequested(response))
-                    .create();
-            dialog.setCanceledOnTouchOutside(false);
-            dialog.show();
+
+            // Do not spam dialogs if this is not the manual update check, use the notifications
+            // instead
+            if (manual) {
+                boolean concat = !response.updateTitle.isEmpty();
+                CharSequence updateMessage =
+                        concat ? TextUtils.concat(response.updateTitle, "; ", response.body) : response.body;
+                AlertDialog dialog = new AlertDialog.Builder(context).setTitle(
+                        getApplicationLabel() + " " + response.versionCodeString + " available")
+                        .setMessage(updateMessage)
+                        .setNegativeButton(R.string.update_later, null)
+                        .setPositiveButton(R.string.update_install, (dialog1, which) -> updateInstallRequested(response))
+                        .create();
+                dialog.setCanceledOnTouchOutside(false);
+                dialog.show();
+            }
+
+            // There is an update, show the notification.
+            //
+            // (In case of the dev build we check whether the apk hashes differ or not beforehand,
+            // so if they are the same this method won't even get called. In case of the release
+            // build this method will be called in both cases so we do the check in this method)
+            notifyNewApkUpdate();
             return true;
         }
+
+        cancelApkUpdateNotification();
         return false;
     }
 
+    private void notifyNewApkUpdate() {
+        ChanSettings.hasNewApkUpdate.set(true);
```

Comment:
This is not a setting. This is a persistable app state. I think I should extract it into a separate class that uses it's own shared preferences file.

Reply:
You probably won't need a setting at all since the update API is run on every startup.

Reply:
But it only checks for the new apk only once in 24h, or is it not?

Reply:
It checks for a new APK only when the app is started cold, with a minimum delay of 24h for dev and 5 days for stable. There's no scheduled task to check for an update, just like how it was in Clover.

Reply:
Well that's why I needed "hasNewApkUpdate" flag. Because if we start the app during the 24h or 5d window when we do not check for the updates, and there were actually an update the last time we checked it, the notification won't show up.

Reply:
Ah okay, I understand. You're setting it in case someone runs the updater manually, but defers installation to a later point.

Reply:
It's still fine to keep it in ChanSettings anyways, there's a bunch of things in there that aren't strictly settings.

Reply:
Well those will have to be moved too. But we can do that later. Or I could move them.

Reply:
Please just keep it to one thing per PR if possible, these super huge PRs with all sorts of barely related stuff in them makes it difficult to review.

Reply:
Alright then I will just move this one setting to a new class.

Reply:
Done.